internal/v2/cache/devices.go

```diff
@@ -0,0 +1,176 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package cache
+
+import (
+	"fmt"
+	"sync"
+
+	edgexErr "github.com/edgexfoundry/go-mod-core-contracts/errors"
+	contract "github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+)
+
+var (
+	dc *deviceCache
```

Comment:
package global is not good practice. Should be stored on the sdk's Service.

Reply:
cache is heavily used by other internal packages, I think we need to put them into dic? This would be a massive refactoring across SDK, how about make the change in next release if available ?

Reply:
Ok, please create a Tech-Debt issue for this.